src/uct/ib/rdmacm/rdmacm_cm.h

```diff
@@ -72,17 +108,31 @@ uct_rdmacm_cm_id_to_dev_name(struct rdma_cm_id *cm_id, char *dev_name)
                       cm_id->port_num);
 }
 
+static UCS_F_ALWAYS_INLINE uint32_t
+uct_rdmacm_cm_get_qpns_per_obj(struct uct_rdmacm_cm_reserved_qpn_context ctx)
+{
+    return 1 << ctx.log_reserved_qpn_granularity;
```

Comment:
maybe save 1 << ctx.log_reserved_qpn_granularity in the context (rename it to qpn_block_size for example)
then, we can define that if qpn_block_size==0 then reserved_qpn is unsupported - and remove reserved_qpn_support field.

Reply:
@yosefe Both (1 << ctx.log_reserved_qpn_granularity) and ctx.log_reserved_qpn_granularity are needed in later code, so we should save one and calculate the other one, or we have to save both. If save one, then better to save ctx.log_reserved_qpn_granularity in the context since is easier to calculate (1 << ctx.log_reserved_qpn_granularity) from ctx.log_reserved_qpn_granularity but the revert is a little more complex.

Reply:
ok, so let's use `UCS_BIT(ctx.log_reserved_qpn_granularity)` instead of this inline function

Reply:
@yosefe Fixed.